### DISCUSSION TITLE
Combine Alpine.data with extra data

### DISCUSSION THREAD

**Discussion by Djules** [alpinejs/alpine#1585] [1. Help] [answered]
Hi, 
I'm looking for a way to combine data from component registered with new `Alpine.data(...)` and extra data which can override the component's one, like props in Vue. It was possible in Alpine v2 but can't find a way to do it in v3.

I used to add an optional argument in my component to override it's data, like:

```html
<div x-data="MyComponent({ title: 'New title' })">
```
```js
window.MyComponent = function ({ settings = {}) {
  return {
    title: 'Default title',
    ...settings
  }
}
```

But it doesn't seem to work with `Alpine.data`.
Can you help?

**Comment by ryangjchandler** [comment, marked as answer]
This currently isn't possible. Ive got a PR open to support it, but I'm going to open another one with an alternative approach too.

**Reply by Djules** [reply]
I just found another discussion (#1521) on this topic, should have check twice. 🙄
